source/server/listener_impl.h

```diff
@@ -224,13 +225,15 @@ class ListenerImpl : public Network::ListenerConfig,
   const uint64_t hash_;
   ProtobufMessage::ValidationVisitor& validation_visitor_;
 
+  // This init watcher, if available, notifies the "parent" listener manager when listener
+  // initialization is complete. It may be reset to cancel interest.
+  std::unique_ptr<Init::WatcherImpl> init_watcher_;
+  // A target is added to Server's InitManager if workers_started_ is false.
+  std::unique_ptr<Init::TargetImpl> listener_init_target_;
```

Comment:
When the `ListenerImpl` is in destroyed, `listener_init_target_` is destroyed prior to `init_watcher_`
Is it possible it triggered init_watcher_ in between?

Reply:
ListenerImpl destruction is supposed to be an "atomic" operation in master thread, same as listener_init_target_'s triggering. I think it's fine here?

Reply:
Sorry for the churn. I agree it's fine. Although for different reason :)
On top of `~ListenerImpl()` there might be `ListenerManagerImpl::startWorkers()` if this listener is the last uninitialized listener. It's fine because this destroying listener won't in the start list

Reply:
ACK.